receiver/haproxyreceiver/config.go

```diff
@@ -0,0 +1,39 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package haproxyreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/haproxyreceiver"
+
+import (
+	"errors"
+
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/config/confighttp"
+	"go.opentelemetry.io/collector/receiver/scraperhelper"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/haproxyreceiver/internal/metadata"
+)
+
+type Config struct {
+	config.ReceiverSettings                 `mapstructure:",squash"`
+	confighttp.HTTPClientSettings           `mapstructure:",squash"`
+	scraperhelper.ScraperControllerSettings `mapstructure:",squash"`
+	MetricsSettings                         metadata.MetricsSettings `mapstructure:"metrics"`
+}
```

Comment:
Can I ask how you intend to support access via a socket? 

I can see how you can reuse some of these values to produce the same effort, I just worry that it may have to change.

Reply:
I think it might be a bit more work to support it, but overall it will still be a http client. Here is a quick example I found:
https://gist.github.com/teknoraver/5ffacb8757330715bcbcc90e6d46ac74

Two changes required, from what I can see. You set up your dial context to use unix mode:

```
DialContext: func(_ context.Context, _, _ string) (net.Conn, error) {
				return net.Dial("unix", flag.Args()[0])
			},
```

You also call out to unix based endpoints:
```
response, err = httpc.Get("http://unix" + flag.Args()[1])
```